library/network/src/lib/y2firewall/firewalld/api.rb

```diff
@@ -0,0 +1,353 @@
+      end
+
+      # @return [Boolean] The firewalld reload result (exit code)
+      def reload
+        return false if offline?
+
+        query("--reload")
+      end
+
+      # @return [Boolean] The firewalld complete-reload result (exit code)
+      def complete_reload
+        return false if offline?
+
+        query("--complete-reload")
+      end
+
+      # @return [Boolean] The firewalld runtime-to-permanent result (exit code)
+      def make_permanent
+        return false if offline?
+
+        query("--runtime-to-permanent")
+      end
+
+      ### Zones ####
+
+      # @return [Array<String>] List of firewall zones
+      def zones
+        query("--get-zones").chomp.split(" ")
+      end
+
+      # @param zone [String] The firewall zone
+      # @return [Array<String>] list of zone's interfaces
+      def list_interfaces(zone)
+        query("--zone=#{zone} --list-interfaces").split(" ")
+      end
+
+      # @param zone [String] The firewall zone
+      # @return [Arrray<String>] list of zone's services
+      def list_services(zone)
+        query("--zone=#{zone} --list-services").split(" ")
+      end
+
+      # @param zone [String] The firewall zone
+      # @return [Array<String>] list of zone's ports
+      def list_ports(zone)
+        query("--zone=#{zone} --list-ports").split(" ")
+      end
+
+      # @param zone [String] The firewall zone
+      # @return [Array<String>] list of zone's protocols
+      def list_protocols(zone)
+        query("--zone=#{zone} --list-protocols").split(" ")
+      end
+
+      # @param zone [String] The firewall zone
+      # @return [Array<String>] list of all information for given zone
+      def list_all(zone)
+        query("--zone=#{zone} --list-all").split(" ")
+      end
+
+      # @return [Array<String>] list of all information for all firewall zones
+      def list_all_zones
+        query("--list-all-zones").split("\n")
+      end
+
+      ### Interfaces ###
+
+      # @param zone [String] The firewall zone
+      # @param interface [String] The network interface
+      # @return [Boolean] True if interface is assigned to zone
+      def interface_enabled?(zone, interface)
+        query("--zone=#{zone} --query-interface=#{interface}")
+      end
+
+      # @param zone [String] The firewall zone
+      # @param interface [String] The network interface
+      # @return [Boolean] True if interface was added to zone
+      def add_interface(zone, interface)
+        query("--zone=#{zone} --add-interface=#{interface}")
+      end
+
+      # @param zone [String] The firewall zone
+      # @param interface [String] The network interface
+      # @return [Boolean] True if interface was removed from zone
+      def remove_interface(zone, interface)
+        query("--zone=#{zone} --remove-interface=#{interface}")
+      end
+
+      ### Services ###
+
+      # @return [Array<String>] List of firewall services
+      def services
+        query("--get-services").to_s.split(" ")
```

Comment:
It's not needed anymore, BTW it probably was added because when it was not running and not called with the `firewall-cmd` Cheetah returns nill because the return code is not 0.